### PR TITLE
refactor(comp:collapse): rebuild with tsx and redesign apis

### PR DESCRIPTION
**Pull request by danranVm**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IDuxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

File: packages/components/collapse/src/CollapsePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      const expanded = isExpanded.value
+      const icon = collapseSlots.expandIcon?.({ key, expanded }) ?? (
+        <IxIcon name={expandIcon.value} rotate={expanded ? 90 : 0}></IxIcon>
+      )
+      const headerSlots = { prefix: () => icon }
+      const { header } = props
+      const headerProps = isString(header) ? { title: header } : header
+      return (
+        <div class={classes.value}>
+          <IxHeader v-slots={headerSlots} size="medium" onClick={handleClick} {...headerProps}></IxHeader>
```
Comment (Usualminds) [inline packages/components/collapse/src/CollapsePanel.tsx:43]: `IxHeader` 还支持自定义 `slot` 不？，当时那个 API 说明是支持用户自定义头部 header

File: packages/components/collapse/src/CollapsePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  props: collapsePanelProps,
+  setup(props, { slots }) {
+    const { slots: collapseSlots, expandedKeys, expandIcon, handleExpand } = inject(collapseToken)!
+
+    const key = useKey()
+    const isExpanded = computed(() => expandedKeys.value.includes(key))
+    const classes = computed(() => {
+      return {
+        'ix-collapse-panel': true,
+        'ix-collapse-panel-disabled': props.disabled,
```
Comment (Usualminds) [inline packages/components/collapse/src/CollapsePanel.tsx:21]: 预览的 `disabled` 样式好像没生效 [collapse disabled](https://preview-431-idux.surge.sh/components/collapse/zh)

File: packages/components/collapse/demo/Basic.vue [hunk 3]
```diff
@@ -1,15 +1,15 @@
 <template>
```
Comment (Usualminds) [inline packages/components/collapse/demo/Basic.vue:1]: 另外几个 demo 用例：嵌套面板、无边框、自定义头部的要同步修改下不，看预览的嵌套面板的效果不太对

**Review by Usualminds — APPROVED** [review]
@danranVm review 完成，提了几个小问题

File: packages/components/collapse/src/CollapsePanel.tsx (hunk 1 above)
Comment (danranVm) [reply packages/components/collapse/src/CollapsePanel.tsx:43]: 给搞忘了，待会加上。

**Pull request merged**